soc/arm/nxp_imx/rt/Kconfig.soc

```diff
@@ -72,6 +72,30 @@ config SOC_MIMXRT1021
 	select HAS_MCUX_EDMA
 	select HAS_MCUX_FLEXCAN
 
+config SOC_MIMXRT1024
+	bool "SOC_MIMXRT1024"
+	select HAS_MCUX
+	select HAS_MCUX_CACHE
+	select HAS_MCUX_CCM
+	select HAS_MCUX_ENET
+	select HAS_MCUX_IGPIO
+	select HAS_MCUX_LPI2C
+	select HAS_MCUX_LPSPI
+	select HAS_MCUX_LPUART
+	select HAS_MCUX_GPT
+	select HAS_MCUX_SEMC
+	select HAS_MCUX_TRNG
+	select CPU_HAS_FPU_DOUBLE_PRECISION
+	select CPU_HAS_ARM_MPU
+	select INIT_SYS_PLL
+	select INIT_USB1_PLL
+	select INIT_ENET_PLL
+	select HAS_MCUX_USB_EHCI
+	select HAS_MCUX_USDHC1
+	select HAS_MCUX_USDHC2
+	select HAS_MCUX_EDMA
+	select HAS_MCUX_FLEXCAN
```

Comment:
You should be careful with selected modules. If the driver is for some of  them still not implemented in Zephyr, better do not choose them. Just follow the scheme of other already implemented NXP socs with similar architecture as yours.